fdbcli/MetaclusterCommands.actor.cpp

```diff
@@ -287,39 +287,80 @@ ACTOR Future<bool> metaclusterStatusCommand(Reference<IDatabase> db, std::vector
 
 	state bool useJson = tokens.size() == 3;
 
-	try {
-		std::map<ClusterName, DataClusterMetadata> clusters =
-		    wait(MetaclusterAPI::listClusters(db, ""_sr, "\xff"_sr, CLIENT_KNOBS->MAX_DATA_CLUSTERS));
-
-		auto capacityNumbers = MetaclusterAPI::metaclusterCapacity(clusters);
-
-		if (useJson) {
-			json_spirit::mObject obj;
-			obj["type"] = "success";
-
-			json_spirit::mObject metaclusterObj;
-			metaclusterObj["data_clusters"] = (int)clusters.size();
-			metaclusterObj["capacity"] = capacityNumbers.first.toJson();
-			metaclusterObj["allocated"] = capacityNumbers.second.toJson();
+	state Reference<ITransaction> tr = db->createTransaction();
 
-			obj["metacluster"] = metaclusterObj;
-			fmt::print("{}\n", json_spirit::write_string(json_spirit::mValue(obj), json_spirit::pretty_print).c_str());
-		} else {
-			fmt::print("  number of data clusters: {}\n", clusters.size());
-			fmt::print("  tenant group capacity: {}\n", capacityNumbers.first.numTenantGroups);
-			fmt::print("  allocated tenant groups: {}\n", capacityNumbers.second.numTenantGroups);
-		}
+	loop {
+		try {
+			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+			Optional<MetaclusterRegistrationEntry> registrationEntry =
+			    wait(MetaclusterAPI::getMetaclusterRegistration(tr));
```

Comment:
Seeing this written, it doesn't seem substantially different from the existing method to read the registration:

```
MetaclusterAPI::getMetaclusterRegistration(tr);
MetaclusterMetadata::metaclusterRegistration().get(tr);
```

How would you feel about removing the new function and using the existing pattern?

Reply:
SGTM